libs/langchain/langchain/document_loaders/googledrive.py

```diff
@@ -290,7 +285,34 @@ def _load_documents_from_ids(self) -> List[Document]:
         if not self.document_ids:
             raise ValueError("document_ids must be set")
 
-        return [self._load_document_from_id(doc_id) for doc_id in self.document_ids]
+        from googleapiclient.discovery import build
+
+        creds = self._load_credentials()
+        service = build("drive", "v3", credentials=creds)
+
+        documents = []
+        for doc_id in self.document_ids:
+            file_data = (
+                service.files()
+                .get(fileId=doc_id, fields="id,mimeType", supportsAllDrives=True)
+                .execute()
+            )
+            documents.extend(self._process_document_by_mimetype(file_data))
```

Comment:
Previously we went straight to `_load_document_from_id` for document types. Now we have some extra overhead where we fetch the file data, pluck out the `id`, and pass the ID to `_load_document_from_id` (which appears to pull the file again).

1. Is this overhead necessary? Will it introduce latency for users?
2. Is the new output from `_load_documents_from_ids` identical to existing supported document types?